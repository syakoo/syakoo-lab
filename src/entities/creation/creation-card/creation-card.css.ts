import { style } from "@vanilla-extract/css";

import { theme } from "@/shared/design-system/theme.css";

export const root = style({
  position: "relative",
  width: "100%",
  minWidth: "150px",
});

export const imageContainer = style({
  position: "relative",
  aspectRatio: "1/1",
  width: "100%",
  overflow: "hidden",
  borderRadius: theme.radius[200],
});

export const image = style({
  objectFit: "cover",
  transition: "transform 0.3s",
});

export const typeContainer = style({
  position: "absolute",
  bottom: theme.space[50],
  right: theme.space[50],
  backgroundColor: "#15212C",
  padding: theme.space[50],
  borderRadius: theme.radius[100],
  display: "flex",
  alignItems: "center",
  justifyContent: "center",
  width: 26,
  height: 26,
});

export const typeIcon = style({
  color: theme.color.text.primary,
  width: "14px",
  height: "14px",
});
