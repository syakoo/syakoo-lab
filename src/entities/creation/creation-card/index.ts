export { CreationCard } from "./creation-card";
