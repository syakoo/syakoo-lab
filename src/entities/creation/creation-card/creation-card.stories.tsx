import type { Meta, StoryObj } from "@storybook/react";

import {
  generateDummyCreationGame,
  generateDummyCreationIllust,
  generateDummyCreationWebapp,
} from "@/entities/creation/models/creation.mocks";

import { CreationCard } from "./creation-card";

const meta = {
  component: CreationCard,
  parameters: {
    layout: "centered",
  },
} satisfies Meta<typeof CreationCard>;

export default meta;
type Story = StoryObj<typeof meta>;

export const Illust: Story = {
  args: { ...generateDummyCreationIllust() },
};

export const Game: Story = {
  args: { ...generateDummyCreationGame() },
};

export const WebApp: Story = {
  args: { ...generateDummyCreationWebapp() },
};
