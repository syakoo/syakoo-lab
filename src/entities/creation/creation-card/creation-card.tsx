import Image from "next/image";
import type { FC } from "react";
import { match } from "ts-pattern";

import type { Creation } from "@/entities/creation/models/creation";
import { Icon } from "@/shared/design-system/icons";
import { Col } from "@/shared/design-system/layout";
import { Text } from "@/shared/design-system/ui";

import * as styles from "./creation-card.css";

export type CreationCardProps = Pick<
  Creation,
  "type" | "title" | "thumbnailSrc"
>;

export const CreationCard: FC<CreationCardProps> = ({
  type,
  title,
  thumbnailSrc,
}) => {
  return (
    <div className={styles.root}>
      <Col gap="50">
        <div className={styles.imageContainer}>
          <Image alt="" className={styles.image} fill src={thumbnailSrc} />
          <div aria-label={type} className={styles.typeContainer}>
            {match(type)
              .with("illust", () => <Icon name="brush" />)
              .with("game", () => <Icon name="game" />)
              .with("webapp", () => <Icon name="globe" />)
              .exhaustive()}
          </div>
        </div>
        <Text as="h3" color="primary" size="75" weight="bold">
          {title}
        </Text>
      </Col>
    </div>
  );
};
