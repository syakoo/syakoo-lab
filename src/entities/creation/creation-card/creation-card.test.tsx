import { render } from "@testing-library/react";

import { CreationCard } from "./creation-card";

describe("CreationCard", () => {
  it("should render correctly", () => {
    const { container } = render(
      <CreationCard
        thumbnailSrc="/test.png"
        title="Test Creation"
        type="illust"
      />,
    );

    expect(container).toMatchSnapshot();
  });
});
