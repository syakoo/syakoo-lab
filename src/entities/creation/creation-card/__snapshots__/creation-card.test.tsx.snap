// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreationCard should render correctly 1`] = `
<div>
  <div
    class="creation-card_root__9id8e30"
  >
    <div
      class="flex_flexStyle__zqca9ti flex_flexStyle_align_stretch__zqca9tu flex_flexStyle_variants_align_stretch__zqca9t9 flex_flexStyle_direction_column__zqca9tk flex_flexStyle_gap_50__zqca9tx flex_flexStyle_variants_gap_50__zqca9tc flex_flexStyle_justify_flexStart__zqca9tl flex_flexStyle_variants_justify_flexStart__zqca9t0"
    >
      <div
        class="creation-card_imageContainer__9id8e31"
      >
        <img
          alt=""
          class="creation-card_image__9id8e32"
          data-nimg="fill"
          decoding="async"
          loading="lazy"
          sizes="100vw"
          src="/_next/image?url=%2Ftest.png&w=3840&q=75"
          srcset="/_next/image?url=%2Ftest.png&w=640&q=75 640w, /_next/image?url=%2Ftest.png&w=750&q=75 750w, /_next/image?url=%2Ftest.png&w=828&q=75 828w, /_next/image?url=%2Ftest.png&w=1080&q=75 1080w, /_next/image?url=%2Ftest.png&w=1200&q=75 1200w, /_next/image?url=%2Ftest.png&w=1920&q=75 1920w, /_next/image?url=%2Ftest.png&w=2048&q=75 2048w, /_next/image?url=%2Ftest.png&w=3840&q=75 3840w"
          style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent;"
        />
        <div
          aria-label="illust"
          class="creation-card_typeContainer__9id8e33"
        >
          <svg
            fill="none"
            stroke="currentColor"
            stroke-width="1.5"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M9.53 16.122a3 3 0 0 0-5.78 1.128 2.25 2.25 0 0 1-2.4 2.245 4.5 4.5 0 0 0 8.4-2.245c0-.399-.078-.78-.22-1.128Zm0 0a15.998 15.998 0 0 0 3.388-1.62m-5.043-.025a15.994 15.994 0 0 1 1.622-3.395m3.42 3.42a15.995 15.995 0 0 0 4.764-4.648l3.876-5.814a1.151 1.151 0 0 0-1.597-1.597L14.146 6.32a15.996 15.996 0 0 0-4.649 4.763m3.42 3.42a6.776 6.776 0 0 0-3.42-3.42"
              stroke-linecap="round"
              stroke-linejoin="round"
            />
          </svg>
        </div>
      </div>
      <h3
        class="text_textStyle__1djq151e text_textStyle_color_primary__1djq151f text_textStyle_variants_color_primary__1djq1510 text_textStyle_size_75__1djq151k text_textStyle_variants_size_75__1djq1515 text_textStyle_weight_bold__1djq151t"
      >
        Test Creation
      </h3>
    </div>
  </div>
</div>
`;
